request/request.go

```diff
@@ -217,6 +217,9 @@ func (r *Request) Scrub(reply *dns.Msg) (*dns.Msg, Result) {
 			re = m - 1
 			continue
 		}
+		if rl == size {
+			break
+		}
 	}
 	// We may come out of this loop with one rotation too many as we don't break on rl == size.
```

Comment:
Does this comment need to be updated?  or is it unrelated to the break on rl == size added above?